src/hm/hmRadio.h

```diff
@@ -105,61 +105,88 @@ class HmRadio : public Radio {
-                while ((micros() - startMicros) < innerLoopTimeout) {  // listen (4088us or?) 5110us to each channel
-                    if (mIrqRcvd) {
-                        mIrqRcvd = false;
+                if (millis() - mTimeslotStart < innerLoopTimeout)
+                    return; // nothing to do
 
-                        if (getReceived()) { // everything received
-                            return;
-                        }
+                // otherwise switch to next RX channel
+                mTimeslotStart = millis();
+                rxPendular = !rxPendular;
+                //innerLoopTimeout = (rxPendular ? 1 : 2)*DURATION_LISTEN_MIN;
+                innerLoopTimeout = DURATION_LISTEN_MIN;
 
-                        innerLoopTimeout = 4088*5;
-                        if (isRxInit) {
-                            isRxInit = false;
-                            if (micros() - startMicros < 42000) {
-                                innerLoopTimeout = 4088*12;
-                                mRxChIdx = (mRxChIdx + 4) % RF_CHANNELS;
-                                mNrf24->setChannel(mRfChLst[mRxChIdx]);
-                            }
-                        }
+                tempRxChIdx = (mRxChIdx + rxPendular*chOffset2) % RF_CHANNELS;
+                mNrf24->setChannel(mRfChLst[tempRxChIdx]);
+                isRxInit = false;
 
-                        startMicros = micros();
-                    }
-                    yield();
+                return;
+            }
+
+            // here we got news from the nRF
+
+            mNrf24->whatHappened(tx_ok, tx_fail, rx_ready); // resets the IRQ pin to HIGH
+            mIrqRcvd   = false;
+
+            if(tx_ok || tx_fail) {                          // tx related interrupt, basically we should start listening
+                mNrf24->flush_tx();                         // empty TX FIFO
+
+                if(mNRFisInRX) {
+                    DPRINTLN(DBG_WARN, F("unexpected tx irq!"));
+                    return;
                 }
-                // switch to next RX channel
-                mRxChIdx = (mRxChIdx + 4) % RF_CHANNELS;
+
+                mNRFisInRX = true;
+                if(tx_ok)
+                    mLastIv->mAckCount++;
+
+                // start listening
+                mRxChIdx = (mTxChIdx + chOffset) % RF_CHANNELS;
                 mNrf24->setChannel(mRfChLst[mRxChIdx]);
-                innerLoopTimeout = 4088;
-                isRxInit = false;
+                mNrf24->startListening();
+                mTimeslotStart = millis();
+                tempRxChIdx = mRxChIdx;
+                chOffset2 = mLastIv->ivGen == IV_HM ? 4 : (mLastIv->mCmd == MI_REQ_CH1 || mLastIv->mCmd == MI_REQ_CH2) ? 1 :4; // reversed channel order for everything apart from 1/2ch MI Data requests
+                rxPendular = false;
+
+                innerLoopTimeout = DURATION_TXFRAME;
+            }
+
+            if(rx_ready) {
```

Comment:
Das würde ich nach oben schieben, das ist m.E. das Timing kritischere und der default ist Empfangen (RX),

Reply:
Kann man machen, aber in meiner idealen Welt gibt es sowieso nur einen der drei Anlässe, nie mehrere gleichzeitig. Von daher war mein Angang für den unwahrscheinlichen Fall, dass doch alles zusammenkommt gewesen, zuerst den wichtigen Teil zu erledigen, und das ist das Zurückschalten auf rx nach tx.